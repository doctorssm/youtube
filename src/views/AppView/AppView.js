export default class AppView {
  constructor(clips) {
    this.clips = clips;
  }

  render() {
    this.renderInput();
    this.renderList(this.clips);
  }

  renderInput() {
    var input = document.createElement("INPUT");
    input.setAttribute("type", "text");

    document.body.appendChild(input);
  }

  renderList(data) {
    const list = document.querySelector('ul');

    if (list) {
      list.remove()
    }

    document.body.appendChild(this.createClipsList(data));
  }

  createClipsList(clips) {
    const list = document.createElement('ul');
    list.innerHTML = clips.map(this.createCard).join('');

    return list;
  }

  createCard({id, snippet}) {
    const { thumbnails, title, description} = snippet;

    return (
      `<li class="card">
        <img class="card-img" src="${thumbnails.medium.url}" alt="Card image cap">
        <div class="card-body">
          <h5 class="card-title">${title}</h5>
          <p class="card-text">${description}</p>
          <a href="https://www.youtube.com/watch?v=${id.videoId}" target="_blank" class="btn btn-primary">Go to video</a>
        </div>
      </li>`
    );
  }
}

