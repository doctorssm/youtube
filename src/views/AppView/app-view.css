ul {
  list-style: none;
}

img {
  width: 100%;
}

h5 {
  font-size: 16px;
  margin-top: 0;

  width: 260px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
}

.card {
  position: relative;
  display: inline-block;
  border: 1px solid lightgray;
  border-radius: 3px;
  margin: 0 10px;
  width: 300px;
}

.card-body {
  padding: 20px;
}

.card-text {
  height: 74px;
  overflow: hidden;
}

.btn {
  display: inline-block;
  text-align: center;
  padding: 10px;
  border-radius: 3px;
  text-decoration: none;
}

.btn-primary {
  color: white;
  background-color: #007bff;
  border-color: #007bff;
}

.btn-primary:hover {
  background-color: #09498d;
  border-color: #09498d;
}
