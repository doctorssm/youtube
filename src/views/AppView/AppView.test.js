import AppView from './AppView';

describe('AppView.prototype.render', () => {
  it('Should be an instance of Function', () => {
    expect(AppView.prototype.render).toBeInstanceOf(Function);
  });

  it('Should be render correctly', () => {
    const context = {
      titles: [
        'Video about JS',
        'Another video ...',
        'I need more video',
        'I need more video',
      ],
    };

    AppView.prototype.render.call(context);

    expect(document.body.innerHTML).toMatchSnapshot();
  });
});
