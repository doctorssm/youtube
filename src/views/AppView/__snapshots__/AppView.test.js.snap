// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppView.prototype.render Should be render correctly 1`] = `"<ul><li>Video about JS</li><li>Another video ...</li><li>I need more video</li><li>I need more video</li></ul>"`;
